### ISSUE TITLE
It cannot generate the images as you did when I reran it

### ISSUE THREAD

**Issue by WinterIsSunny** [utkd/gans#5]
I just ran this code on my gpu but did not get the similar images. actually the generated images are like random noise. And I also tried it on other dataset, but the accuracy of discriminator is always close to 0. I guess it is because of mode collapse, but you have used experience replay. I'm really confused now.

**Comment by abhiishekpal** [comment]
@WinterIsSunny did it finally work for you?
Getting some random noisy output in image.

**Comment by jinglin80** [comment]
The same. Run the code and the generated images are like random noise.

**Comment by utkd** [comment]
There could be multiple things going wrong. Can you give any additional information? What do the losses look like, how many times did you try to run (sometimes it might get stuck in a bad minima), etc?

**Comment by jinglin80** [comment]
My loss is almost zero. Should the loss be defined differently?

**Comment by JohnTailor** [comment]
Same here. Images look like random noise. It might have to do with Keras 2 API, it seems that there is no training going on since epochs run much to fast.

**Comment by RafaelOO** [comment]
I got the same problem. Has anybody found a solution? I do think it has something to do with the keras 2 API. I have checked other models and the same happened.

**Comment by utkd** [comment]
I no longer maintain this work, but since multiple people have commented, I decided to have a look. 
I do not have the exact setup anymore, so I loaded up the DCGAN notebook in Google Colab, downgraded the tensorflow version for compatibility and after running for 30 epochs, this is the output:

![image](https://user-images.githubusercontent.com/2101558/77844233-c0450c00-7159-11ea-8b47-2d63ff791a22.png)

Although the images are blurry, you can still tell they are somewhat like cars and not at all random noise. 
Can you provide more details on what you are running and how? I can try to narrow down the problem.

**Comment by RafaelOO** [comment]
Thank you very much for your answer.

I am also running i t in Google Colab. Instead of downgrading it to 1.15, I modified the imports to run it in tensorflow 2.1. You can see the notebook here: https://drive.google.com/file/d/1RrbPntBjwtmGZ2XkYhDful2I8KGBlsW5/view?usp=sharing

This is an extract of the final output.

Epoch: 1, Generator Loss: 0.0009380715021817013, Discriminator Loss: [0.00207118 0.        ]
![image](https://user-images.githubusercontent.com/7248945/77846893-90c5ec00-71b9-11ea-8917-667d2627067e.png)

Epoch: 2, Generator Loss: 6.536680608114693e-06, Discriminator Loss: [0.00105942 0.        ]
![image](https://user-images.githubusercontent.com/7248945/77846906-9cb1ae00-71b9-11ea-8cef-e02ed6dc6a8a.png)
[...]
Epoch: 15, Generator Loss: 5.7696864306926727e-05, Discriminator Loss: [0.00096073 0.        ]
![image](https://user-images.githubusercontent.com/7248945/77846931-b81cb900-71b9-11ea-8c1e-7ec2b6594569.png)
![image](https://user-images.githubusercontent.com/7248945/77846935-bce16d00-71b9-11ea-9358-c0e5846b113b.png)

**Comment by utkd** [comment]
I tried to run the CGAN notebook and I did get a similar result. I was not able to spend a lot of time to fix this, but it looks like playing with the number of times the generator is trained vs number of time the discriminator is trained has a somewhat positive effect on the samples - even though they are still blurry, but definitely not noisy like above.

For example, train the discriminator once for every N times the generator is trained. You can do this for the generator as well, but it's usually not done. Can you try that out and see if it helps? N is usually in the range 2-5

**Comment by utkd** [comment]
Finally was able to fix the issue - was due to the misplaced 'break' in the experience replay code. I've removed it and ran the notebook on Colab. (Some other minor changes - removed an additional Conv layer in the generator, Train on only 50% of the CIFAR data to speed things up)
The new notebook is now pushed to git.

I'm sorry it took me so long to fix this, but I've been busy with a lot of things at work. Hope you guys  are looking at newer and better GANs by now.
